### PR TITLE
test(react): add useTimeout test case

### LINKED ISSUE #265: [Feature] add test case for useTimeout
@manudeli 

Yes, I agree. The old `useSetTimeout` would fail the test code below. But with the new `useTimeout`, the test passes.

(The test code below is just a reference example, written in `jest` instead of `vitest`. Also, `rerender` in `renderHook` didn't work well with the test, so I redefined the component and utilized `rerender` in `render`).

```tsx
const delayTime = 1000;

const TestComponent = () => {
  const [number, setNumber] = useState(0);

  useTimeout(() => {
    setNumber(number + 1);
  }, delayTime);

  return <div>{number}</div>;
};
```

```tsx
it('should not be re-called even if the component is re-rendered', () => {
  const { rerender } = render(<TestComponent />);

  expect(screen.getByText('0')).toBeInTheDocument();

  act(() => jest.advanceTimersByTime(delayTime));
  expect(screen.getByText('1')).toBeInTheDocument();

  rerender(<TestComponent />);

  act(() => jest.advanceTimersByTime(delayTime))
  expect(screen.getByText('1')).toBeInTheDocument();
});
```

_Originally posted by @ssi02014 in https://github.com/suspensive/react/issues/264#issuecomment-1781310955_

### PR DESCRIPTION
**Pull request by ssi02014**
fix #265 

# Overview
@manudeli 👏

## PR Checklist

- [x] I did below actions if need

1. I read the [Contributing Guide](https://github.com/suspensive/react/blob/main/CONTRIBUTING.md)
2. I added documents and tests.

### CODE REVIEW

File: packages/react/src/hooks/useTimeout.spec.tsx [hunk 1]
```diff
@@ -19,4 +30,18 @@ describe('useTimeout', () => {
     act(() => vi.advanceTimersByTime(MS_100))
     expect(fn).toHaveBeenCalledTimes(1)
   })
+
+  it('should not be re-called even if the component is re-rendered', () => {
```
Comment (manudeli) [inline packages/react/src/hooks/useTimeout.spec.tsx:34]:
```suggestion
  it('should not call callback reveived again even if component using this hook is rerendered', () => {
```

**Review by minsoo-web — APPROVED** [review]
COOL!

**Review by manudeli — APPROVED** [review]

**Comment by manudeli** [comment]
Thanks to @ssi02014 for improving performance of useTimeout

**Comment by ssi02014** [comment]
@manudeli @minsoo-web Thanks!!

**Pull request merged**